nlp/2021_06_18_data_augmentation/totw_nlp_dat_aug.ipynb

```diff
@@ -0,0 +1,6046 @@
+      "metadata": {
+        "id": "gesWuhM9jbLq"
+      },
+      "source": [
+        "## Model "
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "0xqcEiJqBZsB"
+      },
+      "source": [
+        "metric = load_metric(\"accuracy\")\n",
+        "batch_size = 8\n",
+        "epochs = 20\n",
+        "max_steps = epochs * int(((len(book_review_train_ds)*3)/batch_size)) \n",
+        "\n",
+        "run_dicts = [] # list of dicts to store both metrics and logs for all the experiment runs "
+      ],
+      "execution_count": 21,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "o-q6U8DAx0Jp"
+      },
+      "source": [
+        "def compute_metrics(eval_pred):\n",
+        "    \"\"\"\n",
+        "        Calculates the accuracy of the model's predictions, calculated as follows; (TP + TN) / (TP + TN + FP + FN) with TP: True positive TN: True negative FP: False positive FN: False negative\n",
+        "    \"\"\"\n",
+        "\n",
+        "    logits, labels = eval_pred\n",
+        "    predictions = np.argmax(logits, axis=-1)\n",
+        "    return metric.compute(predictions=predictions, references=labels) \n",
+        "\n",
+        "\n",
+        "class LogAccumulatorCallback(TrainerCallback):\n",
+        "    '''\n",
+        "    A class that stores both the training and the evaluation loss\n",
+        "    '''\n",
+        "    \n",
+        "    def __init__(self):\n",
+        "        self.acc_logs = []\n",
+        "\n",
+        "    def on_log(self, args, state, control, logs=None, **kwargs):\n",
+        "        _ = logs.pop(\"total_flos\", None)\n",
+        "        if state.is_local_process_zero and ('loss' in logs or 'eval_loss' in logs):\n",
+        "            self.acc_logs.append(logs.copy())\n",
+        "\n",
+        "\n",
+        "def train_and_evaluate(train_ds, test_ds, identifier):\n",
+        "    def tokenize(batch):\n",
+        "        return tokenizer(batch['text'], padding=True, truncation=True)\n",
+        "    \n",
+        "    train_ds = train_ds.map(tokenize, batched=True, batch_size=len(train_ds))\n",
+        "    test_ds = test_ds.map(tokenize, batched=True, batch_size=len(test_ds))\n",
+        "    \n",
+        "    \n",
+        "    training_args = TrainingArguments(\n",
+        "        \"trainer_args\",\n",
+        "        evaluation_strategy=\"steps\",\n",
+        "        per_device_train_batch_size=batch_size,\n",
+        "        per_device_eval_batch_size=batch_size,\n",
+        "        eval_steps=25,\n",
+        "        logging_steps=25,\n",
+        "        max_steps=max_steps,\n",
+        "        learning_rate=2e-5,\n",
+        "    )\n",
+        "    \n",
+        "    model = AutoModelForSequenceClassification.from_pretrained(\"distilbert-base-multilingual-cased\", num_labels=2)\n",
+        "\n",
+        "    for block in model.distilbert.embeddings.modules():\n",
```

Comment:
Would put a small comment above this to denote what we are doing here (partially freezing) and why